src/main/kotlin/com/google/androidstudiopoet/writers/AbstractWriter.kt

```diff
@@ -14,7 +15,9 @@ interface AbstractWriter {
 
     fun delete(path: String) {
         val file = File(path)
-        file.listFiles()?.forEach { delete(it.absolutePath) }
+        if (file.isDirectory && !FileUtils.isSymlink(file)) {
```

Comment:
Thanks for the PR. Are you sure that you need the check for `isDirectory`?

Reply:
Just noticed that there's a null safety check in `file.listFiles()?` - new to Kotlin :-). Removed.